sycl/include/sycl/reduction.hpp

```diff
@@ -190,6 +190,15 @@ template <class Reducer> class combiner {
   static constexpr size_t Extent = ReducerTraits<Reducer>::extent;
 
 public:
+  // According to SYCL2020 copy constructors for a derived reducer class are
+  // marked as 'delete' to avoid errors when a lambda accepts a private copy of a
+  // `reducer` argument.
+  combiner() {}
```

Comment:
I don't know any guideline, but since the default ctor was implicitly constructed by the compiler, I think it will be better to mark it as default to preserve previous behavior c38ef7d